keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+- Deployer: User who deploys containers.
+    - Deployer Admin: In some scenarios a deployer will include a group of users (i.e. teams or enterprises). Admin users (i.e. security administrators) will be responsible for configuring deployment policies, including potentially a set of trusted roots or required signatures. 
+    - Deployer Operator: In some scenarios a publisher will include a group of users (i.e. teams or enterprises). Operators (i.e. developers, deployment hosts) will be responsible for pulling container images from registries, verifying their signatures, and then running them.
+- Repository Owner: In some scenarios a container may be stored in a repository that is managed by the repository owner. The repository may be public or private and could also be air-gapped.
+
+## Definitions:
+- Root key: A self signed key used for the lowest designation of trust. Root keys can be created by developers, organizations, public/private CAs, and registry operators. The root key should be retrieved from a trusted source that can establish the authenticity of the creator's identity.
+- Signing key: A signing key is used to generate artifact signatures. A signing key should be signed with a root key or one of its intermediaries. The certificate chain with a signing key can be used to verify which root it belongs to. While a root key can be used as a signing key, this is not recommended as it creates a large blast radius and increases the risk of compromising a root key. 
+- Trust Store: The trust store defines the relationship between signing keys and artifacts that are used at validation time to determine whether to trust an artifact with a crytptographically valid signature. The trust store will relate a source (any source, specific registry, specific repository, or specific target) with a certificate (for root key, intermediate, or signing key) or key repository (for automated key distribtuion). It is not recommended to use a signing key as this will cause signature validation to fail if the signing key is rotated. 
+- Trust Store Key Information: The key information configured in the trust store will be cryptographically verifiable and contain:
+    - Public Key: Will be used to verify signed artifacts.
+    - Signed Identity: Identity of creator of the signing key.
+    - (Optional) Issuer: If the key is an intermediate this will describe the owner of the root.
+    - Mechanism to update validity of signatures generated with the signing key.
+
+## Requirements
+- Signing an artifact MUST NOT require the publisher to perform additional actions with a registry/repository or registry operator. Uploading a signature MAY require additional actions.
+- Retrieving a signature MUST NOT require the deployer to perform additional actions with a registry/repository or registry operator beyond those required to pull an unsigned artifact.
```

Comment:
Should we add a requirement for:

A user MUST be able to pull the signature and validate the content before pulling the content? 

This would mitigate trojan horse attacks where the content must be pulled and only later do they determine the content is problematic.

Reply:
Pulling and validating the signature first then pull the content is beneficial in terms of engineering for saving network bandwidth.

However, in some scenarios where the client just want to dump / archive the content and signatures without any validation, they can pull in any order.

Reply:
@shizhMSFT, is there something about the addition of:

> A user MUST be able to pull the signature and validate the content before pulling the content?

That conflicts with the ability to pull the content and signatures? The note above is more related to assuring the signatures are separable from the content, so we can verify the signature before the content. It's not meant to imply a user can't pull content and signatures, just they must be separable.